www/js/share/components/account-upgrade/account-upgrade.controller.js

```diff
@@ -87,31 +87,34 @@
         // check 3 states combining of Maltainvest shortcode, gaming company and financial company
         vm.accountStates = function(landing_company) {
             vm.data.landingCompany = landing_company;
-            if (
-                vm.data.landingCompany.hasOwnProperty("financial_company") &&
-                vm.data.landingCompany.financial_company.shortcode === "maltainvest"
+            if ((vm.data.landingCompany.hasOwnProperty('financial_company') &&
+	            !vm.data.landingCompany.financial_company.shortcode === "maltainvest") ||
+	            !vm.data.landingCompany.hasOwnProperty('financial_company')
             ) {
-                if (vm.data.landingCompany.hasOwnProperty("gaming_company") && vm.data.landingCompany.gaming_company.shortcode === "malta") {
-                    //  check if has MLT then to MF, if not to MLT
-                    if( vm.isVirtual ) {
-                        vm.toReal = true;
-                    } else {
-                        vm.toMaltainvest = true;
-                    }
-                    appStateService.hasMLT = !!vm.toMaltainvest;
-                    vm.getToken();
+	            if( vm.isVirtual ) {
+		            vm.toReal = true;
+		            vm.getToken();
+	            }
+            } else if (vm.data.landingCompany.hasOwnProperty('financial_company') &&
+              vm.data.landingCompany.financial_company.shortcode === "maltainvest" &&
+              vm.data.landingCompany.hasOwnProperty('gaming_company') &&
+                vm.data.landingCompany.gaming_company.shortcode === "malta"
+            ) {
+	             // check if has MLT then to MF, if not to MLT
+                if( vm.isVirtual ) {
```

Comment:
Do you want to check that the user has a virtual account or is sign-in with his/her virtual account?
If you mean user has virtual account, as I check it is not a good way to decide about it because when the user is login as other type of account (other than VRTC) the value of `appStateService.virtuality` is `0`, unless the Open Real Account just is show when the user is sing-in with his/her VRTC account.

Reply:
I want to check if the user is now logged in with VRTC account. for example if user has the option to upgrade to MLT and has no MLT account, he should be logged in with VRTC, then if he upgrades to MLT and he has the option to upgrade to MF, we should only show him the link to upgrade to MF when he is logged in with his MLT account,it means when he is logged in with VRTC he should not see the upgrade link because he already upgraded from VRTC. 
so for this behavior we check what account he's logged in with now

Reply:
So everything seems OK